### PR TITLE
fix: always wait for fluent container removal

### PR DESCRIPTION
**Pull request by stoksc**
## Description
The test flake seen in [here](https://app.circleci.com/pipelines/github/determined-ai/determined/36194/workflows/d3b7286c-a36b-40ab-8163-6cee964348b6/jobs/1299880) shows a two interesting behaviors:
1.  When removal of a container is already in progress, `docker.ContainerRemove` returns an error and does not block until the container is removed. So we can start fluent to bind to ports when it is being removed and perhaps still bound..
2. The fluentbit log stream is started just after the container, meaning we miss logs for instant removals.
3. We start fluentbit and wait for `WaitConditionNotRunning` which works, but if we wait before with `WaitConditionNextExit` we won't miss the exit code. Though 'exit code 1' isn't likely going to be much help.

This fixes 1, which will probably make the flake go away, unless fluent goes into a true crash/backoff loop. Unfortunately, only fixing 2 will likely make the unexplained fluent crashes in these flakes easier to explain, but as far as I can tell you cannot begin tailing logs for a container until it is started (unlike the nice `docker.ContainerWait` + `WaitConditionNextExit`). I can only scheme up using a file logging driver just for fluent. Fixing 3 is a 'nice to have' but I don't think it helps.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: agent/internal/fluent/fluent.go [hunk 1]
```diff
@@ -409,38 +409,73 @@ func (f *Fluent) trackLogs(
...
-	containers, err := docker.Inner().ContainerList(ctx, types.ContainerListOptions{
+	cont, err := containerByName(ctx, dcl, name)
+	switch {
+	case err != nil:
+		return err
+	case cont == nil:
+		return nil
+	}
+
+	log.WithField("id", cont.ID).Infof("removing Docker container %s", name)
```
Comment (stoksc) [inline agent/internal/fluent/fluent.go:423]: prob need to backport this removal code to the docker client, but i think that should be a separate pr

**Review by dzhu — APPROVED** [review]
For 2, can we disable autoremove on the container (maybe just in testing?)
